cdc/sinkv2/eventsink/mq/kafka_sink.go

```diff
@@ -70,7 +75,17 @@ func NewKafkaSink(
 		return nil, errors.Trace(err)
 	}
 
-	producer := newProducerImpl()
+	client, err := sarama.NewClient(baseConfig.BrokerEndpoints, saramaConfig)
+	if err != nil {
+		return nil, cerror.WrapError(cerror.ErrKafkaNewSaramaProducer, err)
+	}
+
+	log.Info("Try to create a producer",
+		zap.Any("baseConfig", baseConfig), zap.Any("saramaConfig", saramaConfig))
```

Comment:
`saramaConfig` contains json unsupported type.

Reply:
I don't understand what you mean. Could you please explain it?

Reply:
Apply the following diff and run case `go test -v -count=1 -run=TestWriteEvents`
```diff
diff --git a/cdc/sinkv2/eventsink/mq/kafka_sink.go b/cdc/sinkv2/eventsink/mq/kafka_sink.go
index 99f95b997..34ee8fee6 100644
--- a/cdc/sinkv2/eventsink/mq/kafka_sink.go
+++ b/cdc/sinkv2/eventsink/mq/kafka_sink.go
@@ -80,7 +80,7 @@ func NewKafkaSink(
                return nil, cerror.WrapError(cerror.ErrKafkaNewSaramaProducer, err)
        }

-       log.Info("Try to create a producer",
+       log.Warn("Try to create a producer",
```

[2022/07/25 14:10:08.975 +08:00] [WARN] [kafka_sink.go:83] ["Try to create a producer"] [baseConfig="{\"BrokerEndpoints\":[\"127.0.0.1:38625\"],\"PartitionNum\":1,\"ReplicationFactor\":1,
\"Version\":\"0.9.0.0\",\"MaxMessageBytes\":1048576,\"Compression\":\"gzip\",\"ClientID\":\"unit-test\",\"EnableTLS\":false,\"Credential\":{\"ca-path\":\"\",\"cert-path\":\"\",\"key-path\
":\"\",\"cert-allowed-cn\":null},\"SASL\":{\"sasl-user\":\"\",\"sasl-password\":\"\",\"sasl-mechanism\":\"\",\"sasl-gssapi\":{\"sasl-gssapi-auth-type\":0,\"sasl-gssapi-keytab-path\":\"\",
\"sasl-gssapi-kerberos-config-path\":\"\",\"sasl-gssapi-service-name\":\"\",\"sasl-gssapi-user\":\"\",\"sasl-gssapi-password\":\"\",\"sasl-gssapi-realm\":\"\",\"sasl-gssapi-disable-pafxfa
st\":false}},\"AutoCreate\":false,\"DialTimeout\":10000000000,\"WriteTimeout\":10000000000,\"ReadTimeout\":10000000000}"] [saramaConfigError="json: unsupported type: func() sarama.SCRAMCl
ient"]

Reply:
Besides running case `TestWriteEvents` alone, I met the following goroutine leak

```golang
--- PASS: TestWriteEvents (1.01s)
[2022/07/25 14:16:51.944 +08:00] [INFO] [worker.go:74] ["flushWorker exited"] [error="context canceled"] [errorVerbose="context canceled\ngithub.com/pingcap/errors.AddStack\n\t/home/apple/go/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/errors.go:174\ngithub.com/pingcap/errors.Trace\n\t/home/apple/go/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/juju_adaptor.go:15\ngithub.com/pingcap/tiflow/cdc/sinkv2/eventsink/mq.(*worker).run\n\t/home/apple/code/tiflow/cdc/sinkv2/eventsink/mq/worker.go:85\ngithub.com/pingcap/tiflow/cdc/sinkv2/eventsink/mq.newSink.func1\n\t/home/apple/code/tiflow/cdc/sinkv2/eventsink/mq/mq_sink.go:89\nruntime.goexit\n\t/home/apple/.g/go/src/runtime/asm_amd64.s:1571"] [namespace=] [changefeed=]
PASS
goleak: Errors on successful test run: found unexpected goroutines:
[Goroutine 23 in state IO wait, with internal/poll.runtime_pollWait on top of the stack:
goroutine 23 [IO wait]:
internal/poll.runtime_pollWait(0x7f9f7c5f0028, 0x72)
        /home/apple/.g/go/src/runtime/netpoll.go:302 +0x89
internal/poll.(*pollDesc).wait(0xc000288200?, 0xc000556470?, 0x0)
        /home/apple/.g/go/src/internal/poll/fd_poll_runtime.go:83 +0x32
internal/poll.(*pollDesc).waitRead(...)
        /home/apple/.g/go/src/internal/poll/fd_poll_runtime.go:88
internal/poll.(*FD).ReadMsg(0xc000288200, {0xc000556470, 0x10, 0x10}, {0xc00055a020, 0x1000, 0x1000}, 0x16a950f?)
        /home/apple/.g/go/src/internal/poll/fd_unix.go:304 +0x31c
net.(*netFD).readMsg(0xc000288200, {0xc000556470?, 0x7f9f7ce11338?, 0x5?}, {0xc00055a020?, 0x16a81e5?, 0x6fb3800?}, 0xc00056fd08?)
        /home/apple/.g/go/src/net/fd_posix.go:78 +0x37
net.(*UnixConn).readMsg(0xc0004fa8d0, {0xc000556470?, 0xc00056fd80?, 0x16a12e5?}, {0xc00055a020?, 0xb?, 0x7f9f7eb13fff?})
        /home/apple/.g/go/src/net/unixsock_posix.go:115 +0x4f
net.(*UnixConn).ReadMsgUnix(0xc0004fa8d0, {0xc000556470?, 0xc00056fdf8?, 0x16a12e5?}, {0xc00055a020?, 0x0?, 0x0?})
        /home/apple/.g/go/src/net/unixsock.go:143 +0x3c
github.com/godbus/dbus.(*oobReader).Read(0xc00055a000, {0xc000556470?, 0xc00056fe20?, 0x16a1647?})
        /home/apple/go/pkg/mod/github.com/godbus/dbus@v0.0.0-20190726142602-4481cbc300e2/transport_unix.go:21 +0x45
io.ReadAtLeast({0x4c505e0, 0xc00055a000}, {0xc000556470, 0x10, 0x10}, 0x10)
        /home/apple/.g/go/src/io/io.go:331 +0x9a
io.ReadFull(...)
        /home/apple/.g/go/src/io/io.go:350
github.com/godbus/dbus.(*unixTransport).ReadMessage(0xc000128ea0)
        /home/apple/go/pkg/mod/github.com/godbus/dbus@v0.0.0-20190726142602-4481cbc300e2/transport_unix.go:91 +0x11e
github.com/godbus/dbus.(*Conn).inWorker(0xc0001f97a0)
        /home/apple/go/pkg/mod/github.com/godbus/dbus@v0.0.0-20190726142602-4481cbc300e2/conn.go:294 +0x3b
created by github.com/godbus/dbus.(*Conn).Auth
        /home/apple/go/pkg/mod/github.com/godbus/dbus@v0.0.0-20190726142602-4481cbc300e2/auth.go:118 +0x9ed

]
exit status 1
FAIL    github.com/pingcap/tiflow/cdc/sinkv2/eventsink/mq       1.545s
```

Reply:
Removed. 
Thanks for your review!  💚 💙 💜 💛 ❤️